shared-bindings/displayio/Group.c

```diff
@@ -328,6 +328,21 @@ STATIC mp_obj_t group_subscr(mp_obj_t self_in, mp_obj_t index_obj, mp_obj_t valu
     return mp_const_none;
 }
 
+//|     def sort(self, key: function, reverse: bool) -> None:
+//|         """Sort the members of the group."""
+//|         ...
+//|
+STATIC mp_obj_t displayio_group_obj_sort(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    displayio_group_t *self = native_group(pos_args[0]);
+    mp_obj_t *args = m_new(mp_obj_t, n_args);
+    for (size_t i = 1; i < n_args; ++i) {
+        args[i] = pos_args[i];
+    }
+    args[0] = self->members;
+    return mp_obj_list_sort(n_args, pos_args, kw_args);
```

Comment:
Shouldn't this call into shared-module as well? As-is, it won't trigger any rectangular redraws.

It may be worth comparing the previous sort vs current to limit what area is updated too.

Reply:
This is a bit tricky to do. We would need to find all descendants that moved to the front of any descendants the overlapped with.

For the particular use case I need, the sprites being sorted are animated and thus will be refreshed on every frame anyways.

I wonder if we could simply mark the whole group as dirty?

Reply:
Thinking further, we would need to make a copy of the original ``members`` before sorting, and then, after sorting, iterate over it, and for every element, locate its previous position — if that position is more to the back than the new one, then iterate over all layers in between the two positions, and check for overlaps — if an overlap is detected, mark the layer dirty.

A simplified, and possibly much faster approach, would be to mark dirty any layers that moved forward, no matter if they overlapped anything or not.

Reply:
This is actually much more complex, as a layer can stay in the same position in the list, and still require redrawing if something moved behind it.

Reply:
I'm tempted to just do bubble sort, and check for overlap each time we swap two items.

Reply:
I wonder if it matters. The sorting is likely to coincide with a tilegrid moving and it'll cause a redraw of the area anyway.

Reply:
I will leave it as it is for now, and see if I can improve it later.